frontend/tests/README.md

```diff
@@ -0,0 +1,121 @@
+# Tests documentation
+
+## How to run E2E tests locally
+
+### What
+
+Running E2E tests locally means the tests are run in your local machine against the web app that's either running locally as well (localhost, docker) OR a deployed environment (dev, staging, prod)
+
+### When
+
+There are a few scenarios you might want to run tests locally:
+
+#### Against local app (`make local-init` or `npm run dev`)
+
+1. You are preparing a new PR and you want to make sure all existing tests pass, and/or you want to add new tests to cover your new feature or bug fix
+
+2. A PR E2E test failed, and you want to see if the test failure is environment dependent.
+
+   E.g., there are two ways to run your FE app locally, either in a Docker container through make commands (`make local-init`) or straight up using `npm run dev` in the `/frontend` directory. PR E2E test runs against local app in a Docker container, so if you run local tests against `npm run dev` started local app and everything is passing, but failing against `make` started Docker container local app, then likely the issue is related to the FE app inside a Docker container.
+
+3. A GHA E2E test against a deployed environment (dev, staging, prod) failed, and you want to see if the test failure is environment dependent.
+
+   E.g., when a test fails against a deployed environment, it's helpful to run the same test locally against your local app to see if it's passing. Because if it is passing, the test failure could be environment dependent, such as env dependent data, data size, AWS machine size, third party rate limiting and/or outage
+
+#### Against a deployed app (dev, staging, prod, rdev)
+
+1. A GHA E2E test against a deployed environment failed, and you want to see if it's just flaky.
+
+   E.g., since tests run a lot slower in GHA vs. locally due to several reasons, such as not as powerful machine, long GHA test queue, and that GHA can't run tests in parallel in the same machine (not many CPUs to utilize), it's sometimes just a lot faster to run that one failed test locally against the same deployed environment
+
+2. A GHA E2E test against a deployed environment failed, and you want to debug the test code.
+
+   E.g., sometimes a test failure is only reproducible in a deployed environment, in such case, you have no choice but to debug the test against the deployed environment
+
+### Where
+
+All the E2E test commands can be found in `frontend/Makefile` and `frontend/package.json`. The `frontend/Makefile` commands are wrappers of `frontend/package.json` commands, so you can use either.
```

Comment:
what do you think of cleaning up the makefile / package.json a bit actually? there are a few commands that seem redundant and a bit confusing to me at a glance (i.e. what is 'smoke-test-prod-build' vs e2e-prod?)

Reply:
Also happy to make this a separate follow-up ticket if you agree it makes sense but prefer to tackle it separately